### PR TITLE
Fetch GitHub repos using the GitHub v4 API

### PR DESCRIPTION
**Pull request by dandelionmane**
Summary:
It’s a whole new world of GraphQL! Our parser is now just a GraphQL
query that asks for exactly what we want and dumps it to a file. The
data exposed by the v4 API is also in a much nicer format than that of
the v3 API, so this is pretty much a universal improvement.

Currently, we do not handle pagination. We require that the repository
in question have fewer than a fixed number of issues, and comments per
issue, and reviews per PR, and review comments per PR, and so on. If
this limit is exceeded, the script will fail-fast with a nice error
message. To fix this, we’ll need to write a general-purpose pagination
API that allows traversing cursors at any level of the query.

Paired with @wchargin.

Test Plan:
Run

    $ GITHUB_TOKEN="your_token_here" src/backend/fetchGitHubRepoTest.sh

and verify that it exits with 0. Note that if you change this script’s
repository from `tiny-example-repository` to `sourcecred`, the script
correctly fails and outputs a useful diff.

wchargin-branch: github-v4-graphql

### CODE REVIEW

File: src/backend/fetchGitHubRepo.js [hunk 1]
```diff
@@ -12,119 +12,168 @@ const octokitFactory = require("@octokit/rest");
...
-  fetchPullRequests() {
-    return this.paginate(
-      this._octokit.pullRequests.getAll({
-        owner: this._repoOwner,
-        repo: this._repoName,
-        state: "all",
-        per_page: 100,
-      })
-    );
+  const query = `query {
```
[the remark this answers is not in the document]
Comment (dandelionmane) [reply src/backend/fetchGitHubRepo.js]: Done.

File: src/backend/fetchGitHubRepoTest.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -eu
+
+canonicalize() {
+  script="$(cat <<EOF
```
[the remark this answers is not in the document]
Comment (dandelionmane) [reply src/backend/fetchGitHubRepoTest.sh]: I removed the canoncialize function per your suggestion below.

File: src/backend/fetchGitHubRepoTest.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -eu
+
+canonicalize() {
```
[the remark this answers is not in the document]
Comment (dandelionmane) [reply src/backend/fetchGitHubRepoTest.sh]: Done. Added a dep on json-stable-stringify.

File: src/backend/fetchGitHubRepo.js [hunk 4]
```diff
@@ -12,119 +12,168 @@ const octokitFactory = require("@octokit/rest");
...
+    repository(owner: "${repoOwner}", name: "${repoName}") {
```
[the remark this answers is not in the document]
Comment (dandelionmane) [reply src/backend/fetchGitHubRepo.js]: Cool!

File: package.json [hunk 5]
```diff
@@ -30,8 +29,10 @@
     "html-webpack-plugin": "2.29.0",
     "husky": "^0.14.3",
     "jest": "20.0.4",
+    "json-stable-stringify": "^1.0.1",
```
Comment (wchargin) [inline package.json]: Great; this was the package that I was looking at, too.

**Pull request merged**